### PR TITLE
dynamic column widths; scroll many results

### PR DESCRIPTION
**Pull request by jtrein**
merge #15 first and this should look cleaner in the commits below. This is based off of #15 for some other things it provides.

**dynamic column widths**
* columns were difficult to resize without going into sussol-react-table, so this takes care of a simple, dynamic resize.
* columns were static, otherwise - yuck!

**Now scrolls whatever amount of results**

### CODE REVIEW

File: src/js/components/Search.jsx [hunk 1]
```diff
@@ -24,11 +54,16 @@ export class Search extends PureComponent {
           onSearchChange={(data, searchTerm) => (
             this.handleSearchChange(data, searchTerm)
           )}
+          onSearchClear={() => this.handleSearchChange(null)}
         />
         {/* render results table */}
         {this.state.showResults && <ResultsTable
           {...props}
+          changeColumnWidths={(columns, table) => this.handleCalculateColumnWidths(columns, table)}
```
Comment (Chris-Petty) [inline src/js/components/Search.jsx]: You are providing this prop, but I don't see in blueprintjs table or in the sussol table wrapper any use of this `changeColumnWidths`.

Comment (Chris-Petty) [inline src/js/components/Search.jsx]: perhaps you mean to be using the prop `onColumnWidthChanged`.

http://blueprintjs.com/docs/#components.table-js.api.table

File: src/js/components/Search.jsx [hunk 2]
```diff
@@ -6,16 +6,46 @@ import { SearchBar } from './SearchBar.jsx';
...
   // respond to search input
   handleSearchChange(data, searchTerm) {
     if (!data || !data.length) this.setState({ showResults: false });
     this.setState({ showResults: true, resultData: data, searchTerm });
   }
 
+  /**
+  * handleCalculateColumnWidths
+  *  - provides naïve, but sane, column widths
+  *  - currently, simply divides columns into equal widths
```
Comment (Chris-Petty) [inline src/js/components/Search.jsx]: Is the built in column width adjustment not satisfactory? For sanity, there are the props `maxColumnWidth` and `minColumnWidth` for limiting how wild the user can be.

File: src/js/components/Search.jsx [hunk 3]
```diff
@@ -6,16 +6,46 @@ import { SearchBar } from './SearchBar.jsx';
...
+  * @param {columns}    columns - object of column data from Blueprint table
+  * @param {obj}        table - referenced table DOM element
+  * @returns {arr|null} array of new widths, or null
+  */
+  handleCalculateColumnWidths(columns, table) {
+    if (columns === undefined || table === undefined) return null;
+    // init array
+    const columnWidths = [0, 0];
+    const tableWrapWidthDivided = table.offsetWidth / columns.length;
+    for (let i = 0; i < columnWidths.length; i += 1) {
```
Comment (jtrein) [inline src/js/components/Search.jsx]: you can thank Airbnb

File: src/js/components/Search.jsx (hunk 1 above)
Comment (jtrein) [inline src/js/components/Search.jsx]: I see. Right, this is a property on *our* component, not SussolTable

File: src/js/components/Search.jsx (hunk 2 above)
Comment (jtrein) [inline src/js/components/Search.jsx]: 0o0o when I scanned the docs I didn't think much about it. Let me check to see if it automagically expands/contracts the columns...

Comment (jtrein) [inline src/js/components/Search.jsx]: doesn't work ❌. The real way to do it is to add support from the sussol-react-table API.  Most likely adding width/height properties to the Cell component using Blueprint's `renderCell` would be in order; though don't hold me to the speculation.

File: src/js/components/Search.jsx (hunk 1 above)
Comment (jtrein) [inline src/js/components/Search.jsx]: See ResultsTable.jsx:9

File: src/js/components/Search.jsx (hunk 3 above)
Comment (Chris-Petty) [inline src/js/components/Search.jsx]: TIL

Comment (Chris-Petty) [inline src/js/components/Search.jsx]: We must have disabled it on Mobile repo

File: src/js/components/Search.jsx (hunk 1 above)
Comment (Chris-Petty) [inline src/js/components/Search.jsx]: Ah yea I see. In that case can you just add it to propTypes (eslint airBnB rules should be throwing a wobbly that they aren't defined) of <ResultTable>... This should do it:

```
ResultTable.propTypes = {
  ...SussolReactTable.propTypes
  changeColumnWidths: React.PropTypes.func,
};
```

Comment (jtrein) [inline src/js/components/Search.jsx]: `...SussolReactTable.propTypes` doesn't SussolReactTable ensure its own PropTypes? Anyway, didn't know you could spread syntax like that!

Comment (jtrein) [inline src/js/components/Search.jsx]: ok, pushed the changes

**Pull request merged**